examples/While.bgl

```diff
@@ -28,3 +28,115 @@ incr(x) = x + 1
 
 whileTest : Int -> Int
 whileTest(x) = while lessThanTen(x) do incr(x)
+
+false : Bool
+false = let b = True in while b do False
+
+-- a while loop that gets its local, mutable context from a function
+toTenF : Int -> Int
+toTenF(x) = while x < 10 do x + 1
```

Comment:
> This is a bit weird because there's no explicit secondary branch

That's true. We could probably discuss it more as a group. Some advantages of excluding an explicit else case:

- it ensures that our `while` closely resembles the typical imperative formulation, where the "else case" is always to do nothing
- most loops will have `else <return state unchanged>`. The current syntax removes this boilerplate
- it completely eliminates this class of bugs:
  - `let x = <expr> in while <condition involving x> do <expr involving x> else y`
- if the else case is more complicated, it can be easily specified with the current syntax:
  - `let x = <expr> in while <c> do <b> else x + 2` =>  `(let x = <expr> in while <c> do <b>) + 2`